pkg/grpcfactory/server.go

```diff
@@ -32,7 +34,17 @@ type COSIProvisionerServer struct {
 	listenOpts        []grpc.ServerOption
 }
 
-func (s *COSIProvisionerServer) Run(ctx context.Context) error {
+func (s *COSIProvisionerServer) Run(ctx context.Context, registry prometheus.Registerer) error {
+
+	srvMetrics := grpcprom.NewServerMetrics(
+		grpcprom.WithServerHandlingTimeHistogram(
+			grpcprom.WithHistogramBuckets([]float64{0.001, 0.01, 0.1, 0.3, 0.6, 1, 3, 6, 9, 20, 30, 60, 90, 120}),
```

Comment:
Note that histogram can greatly increase cardinality.

If this runs on client side outside our control, maybe it could be interesting to have those buckets configurable so they can change easily is they want more or less cardinality or if they want to target more specifically a certain range

Reply:
After re-evaluating i [changed](https://github.com/scality/cosi-driver/pull/83/commits/8cc730891878aa9154d6d951db8032698b2aa2e5) the bucket size and also added [a ticket](https://scality.atlassian.net/browse/S3C-9690) to make it configurable in the future.